src/contexts/application-context/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { createContext, useEffect, useState } from 'react';
+import type { ApplicationContextParams, ApplicationProviderProps } from './types';
+import useCookie from 'react-use-cookie';
```

Comment:
nice library. It's not harmful to learn how cookies are handled natively without 3rd parties, just for learning